### PR TITLE
upgrade to net8.0

### PR DESCRIPTION
**Pull request by SebastianWachsmuth**

### CODE REVIEW

File: src/Blazor.Diagrams.Algorithms/Blazor.Diagrams.Algorithms.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (robertmclaws) [inline src/Blazor.Diagrams.Algorithms/Blazor.Diagrams.Algorithms.csproj:4]: This should be using `<TargetFrameworks>net8.0;net7.0;net6.0;</TargetFrameworks>` instead. The package should compile to all current .NET runtimes.

File: src/Blazor.Diagrams.Core/Blazor.Diagrams.Core.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		<TargetFramework>net6.0</TargetFramework>
+		<TargetFramework>net8.0</TargetFramework>
```
Comment (robertmclaws) [inline src/Blazor.Diagrams.Core/Blazor.Diagrams.Core.csproj:4]: This should be using `<TargetFrameworks>net8.0;net7.0;net6.0;</TargetFrameworks>` instead. The package should compile to all current .NET runtimes.

File: src/Blazor.Diagrams/Blazor.Diagrams.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
 
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
```
Comment (robertmclaws) [inline src/Blazor.Diagrams/Blazor.Diagrams.csproj:4]: This should be using `<TargetFrameworks>net8.0;net7.0;net6.0;</TargetFrameworks>` instead. The package should compile to all current .NET runtimes.

File: src/Blazor.Diagrams/Blazor.Diagrams.csproj [hunk 4]
```diff
@@ -20,8 +20,8 @@
     </PropertyGroup>
 
     <ItemGroup>
```
Comment (robertmclaws) [inline src/Blazor.Diagrams/Blazor.Diagrams.csproj:22]: You're going to need to make the references conditional on the active runtime. See https://github.com/CloudNimble/BlazorEssentials/blob/main/src/CloudNimble.BlazorEssentials/CloudNimble.BlazorEssentials.csproj#L38-L66

**Comment by SebastianWachsmuth** [comment]
I added Multitargetting to the library project, but left the demo and docs at net8

**Comment by robertmclaws** [comment]
Nice! Only think I would say from here, and this is super nit-picky I know, is to go into the .csproj and .props files and Ctrl+e, D the documents so that the indents are consistent.

**Comment by jklemmack** [comment]
@robertmclaws Is there anything still holding up this PR?  I have an interest in using a net8 version directly.  Thanks!

**Comment by robertmclaws** [comment]
I think it looks great and can be merged.

**Comment by zHaytam** [comment]
Hello, thanks for the PR!

I just learned a new thing! Never knew about the props file... I will test this locally quickly and merge asap, thanks again

**Pull request merged**